components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3641,6 +3641,17 @@ public boolean isConfigured() {
         return certificateManager.isConfigured();
     }
 
+    @Override
+    public CertificateMetadataDTO getCertificate(String userName, String alias) throws APIManagementException {
+        int tenantId = 0;
+        try {
+            tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(tenantDomain);
+        } catch (UserStoreException e) {
+            handleException("Error while reading tenant information", e);
```

Comment:
Give more context to the error message, eg: Error while retrieving the tenant information for tenantDomain + tenantDomain

Reply:
This was the error message given in similar methods/contexts in the current APIM implementation